app/controllers/api/account.php

```diff
@@ -940,12 +943,16 @@
     ->inject('locale')
     ->inject('queueForEvents')
     ->inject('queueForMails')
-    ->action(function (string $userId, string $email, string $url, Request $request, Response $response, Document $user, Document $project, Database $dbForProject, Locale $locale, Event $queueForEvents, Mail $queueForMails) {
+    ->action(function (string $userId, string $email, string $url, bool $securityPhrase, Request $request, Response $response, Document $user, Document $project, Database $dbForProject, Locale $locale, Event $queueForEvents, Mail $queueForMails) {
 
         if (empty(App::getEnv('_APP_SMTP_HOST'))) {
             throw new Exception(Exception::GENERAL_SMTP_DISABLED, 'SMTP disabled');
         }
 
+        if ($securityPhrase === true) {
+            $securityPhrase = SecurityPhrase::generate();
+        }
+
```

Comment:
I think we might have an issue with boolean parameters in some SDKs. 

See what we do for the activate param in functions. I believe its still broken ?
https://github.com/appwrite/appwrite/blob/7e1b618769448fd222d51551f64080d49443c1fe/app/controllers/api/functions.php#L1067

Reply:
Problem should only be there with GET requests, and non-JSON body requests. It also relates to `new Boolean()` being `loose: true`.

Here we have loose false (default), so we can always trust it to be boolean from JSON body (its POST request)

Reply:
The functions request is also a POST request. 
Have you tested your endpoint with JS? We noticed this issue in the console so it may be a JS specific thing.

Reply:
Function request is file upload. Its not JSON body.

Reply:
Ill test with JS SDK to be extra safe

Reply:
the check `$securityPhrase === true` will probably always be false. because the parameter `$securityPhrase` may be processed as a string by the server

Reply:
Yea what I mean to say is, JS ( and maybe more languages ) send parameters in a POST request body as strings. So the validator itself will block those requests

Reply:
There are other places such as `required` on new string attribute, or `logging` on function update. There we do not convert booleans. If this was a problem, we would have noticed it by now. What we noticed so far is only new deployment, which is not JSON body.

I did QA with web SDK and it works as expected.

![CleanShot 2024-01-11 at 10 55 37@2x](https://github.com/appwrite/appwrite/assets/19310830/8a707324-8c84-46b3-9f0d-0552b0dcf279)

![CleanShot 2024-01-11 at 10 55 45@2x](https://github.com/appwrite/appwrite/assets/19310830/8b31e2da-e3a2-4f28-b3bc-c9627697503f)

Reply:
What happens when you pass 'true' or 'false' as a parameter ? In both cases I would expect securityPhrase to be empty

Reply:
You get yelled at by IDE, since type is `securityPhrase?: boolean | undefined`.
Also by validator, you get 400 `Invalid `securityPhrase` param: Value must be a valid boolean`.

If you pass string there, you are not using SDKs correctly.